Add below files for telco-hub 4.18

mgonzalezo opened the pull request:
- Add metadata.yaml with 4.18-specific telco-hub component definitions
- Add ReferenceVersionCheck.yaml for OpenShift 4.18 version validation
- Add version_match.tmpl template function for version checking
- Enables automated configuration drift detection for telco-hub deployments

Resolves missing kubectl cluster-compare functionality for telco-hub reference design.

mgonzalezo commented:
@imiller0 @lack looking for some feedback on this.

lack commented:
/ok-to-test

lack commented:
I think you should probably copy over not just the metadata.yaml, but the entire tree under https://github.com/openshift-kni/telco-reference/tree/main/telco-hub/configuration/reference-crs-kube-compare

Reason being that while the cluster-compare tool can operate with plain CRs as input, it would flag any differences from these hard-coded defaults as deviations, which isn't often the goal of setting up cluster-compare reference templates.

For example, take a look at https://github.com/openshift-kni/telco-reference/blob/release-4.18/telco-hub/configuration/reference-crs/required/gitops/ztp-installation/policies-app.yaml -> Without templating, if the user changes the repoUrl to their own gitops repository, it will show up as a difference.

Compare this to the corresponding template in the main branch https://github.com/openshift-kni/telco-reference/blob/main/telco-hub/configuration/reference-crs-kube-compare/required/gitops/argocd-application.yaml that has a templated value for repoUrl which allows customization of that field without the tool flagging it as a difference.

mgonzalezo commented:
/ok-to-test

imiller0 commented on telco-hub/configuration/reference-crs/ReferenceVersionCheck.yaml:
Cluster compare CRs should not be included in the reference-crs tree.

mgonzalezo replied:
Understood. File removed.

imiller0 commented on telco-hub/configuration/reference-crs/metadata.yaml:
Cluster compare CRs should not be included in the reference-crs tree.

mgonzalezo replied:
Understood. File removed.

mgonzalezo commented:
Hi @imiller0 , 
Addressed your feedback on PR #331:
- Removed cluster-compare files from reference-crs/
- Squashed commits for cleaner history.

Ready for your review.
